src/App.tsx

```diff
@@ -1,12 +1,93 @@
-import React from 'react';
+import { Component } from 'react';
+import { List } from './components/List';
+import { LoadingError } from './components/LoadingError';
 import './App.scss';
-
-// import { getAll, get5First, getRed } from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Loader } from './components/Loader';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goodsList: Good[];
+  isStarted: boolean;
+  loading: boolean;
+  error: boolean;
+}
+
+class App extends Component<{}, State> {
+  state: State = {
+    goodsList: [],
+    isStarted: false,
+    loading: false,
+    error: false,
+  };
+
+  displayList = async () => {
+    this.setState(state => ({
+      isStarted: !state.isStarted,
+    }));
+
+    await this.loadGoods(getAll);
+  };
+
+  loadGoods = async (goodsFromServer: () => Promise<Good[]>) => {
+    this.setState({ error: false, loading: true });
+    try {
+      const goods = await goodsFromServer();
+
+      this.setState({ goodsList: goods, error: false, loading: false });
+    } catch {
+      this.setState({ error: true });
```

Comment:
If error occurs, the loading will be infinite, you should change `loading` value here